pkg/trace/config/apply.go

```diff
@@ -111,6 +112,10 @@ func (c *AgentConfig) applyDatadogConfig() error {
 	}
 	if config.Datadog.IsSet("api_key") {
 		c.Endpoints[0].APIKey = config.Datadog.GetString("api_key")
+	} else if apiKeyFile := config.Datadog.GetString("api_key_file"); apiKeyFile != "" {
+		if apiKey, err := ioutil.ReadFile(apiKeyFile); err == nil {
+			c.Endpoints[0].APIKey = string(apiKey)
+		}
```

Comment:
Here api_key has clearly precedence over api_key_file.

Reply:
I’ve reversed the precedence to be more consistent.
Now, `api_key_file` should have precedence over `api_key` everywhere.
Setting simultaneously the two isn’t useful but having a consistent behaviour is important. There is no obvious reason for choosing one precedence order rather than the other. So, I chose the precedence order that we had in agent 5: DataDog/docker-dd-agent#208.